src/xmpp/presence.c

```diff
@@ -656,6 +658,69 @@ _available_handler(xmpp_stanza_t* const stanza)
 
     if (g_strcmp0(xmpp_presence->jid->barejid, my_jid->barejid) == 0) {
         connection_add_available_resource(resource);
+        char* account_name = session_get_account_name();
+        int max_sessions = accounts_get_max_sessions(account_name);
+        char* cur_resource = accounts_get_resource(account_name);
+        if (max_sessions > 0) {
+            int res_count = connection_count_available_resources();
+            if (res_count > max_sessions && g_strcmp0(cur_resource, resource->name)) {
+                ProfWin* console = wins_get_console();
+                ProfWin* current_window = wins_get_current();
+                auto_gchar gchar* message = g_strdup_printf("Max sessions alarm! (%d/%d devices in use)", res_count, max_sessions);
+                win_println(console, THEME_RED, "|", message);
```

Comment:
```suggestion
                win_println(console, THEME_RED, "|", "%s", message);
```

This string is statically created and has no chance of format string injection, but good practice is to only use const format strings.

@jubalh one easy improvement of the code base security-wise could be to use [`__attribute__ ((format (printf, x, y)))`](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-format-function-attri) in our `printf` class of internal API. What do you think?

Reply:
I didn't quite understand what it does. Will read the docu again later :D